src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiResponseTypeProvider.cs

```diff
@@ -56,11 +56,11 @@ public ICollection<ApiResponseType> GetApiResponseTypes(ControllerActionDescript
             return apiResponseTypes;
         }
 
-        private IReadOnlyList<IApiResponseMetadataProvider> GetResponseMetadataAttributes(ControllerActionDescriptor action)
+        private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(ControllerActionDescriptor action)
         {
             if (action.FilterDescriptors == null)
             {
-                return Array.Empty<IApiResponseMetadataProvider>();
+                return new List<IApiResponseMetadataProvider>(capacity: 0);
```

Comment:
specifying a capacity of 0 doesn't do anything, and it possibly hurts. Lists constructed without setting a capactity always start with a shared empty array and only allocate storage when you add something.